pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/JavaInstance.java

```diff
@@ -155,10 +155,24 @@ private void processAsyncResults(JavaInstanceRunnable.AsyncResultConsumer result
 
     }
 
+    public void initialize() throws Exception {
+        if (function != null) {
+            function.initialize(context);
+        }
+    }
+
     @Override
     public void close() {
         context.close();
         executor.shutdown();
+
+        if (function != null) {
```

Comment:
I think we should first close the function resources before close context and executor.